test/gateway/staker_proxy/stake.js

```diff
@@ -0,0 +1,191 @@
+// Copyright 2019 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+'use strict';
+
+const SpyGateway = artifacts.require('./SpyEIP20Gateway.sol');
+const SpyToken = artifacts.require('./SpyToken.sol');
+const StakerProxy = artifacts.require('./StakerProxy.sol');
+const Utils = require('../../test_lib/utils');
+
+contract('StakerProxy.stake()', (accounts) => {
+  const [composer, owner, beneficiary, someoneElse] = accounts;
+
+  const amount = '10001';
+  const gasPrice = '400';
+  const gasLimit = '10000000';
+  const nonce = '42';
+  const hashLock = '0x6666666666666666666666666666666666666666666666666666666666666666';
+
+  let stakerProxy;
+  let spyGateway;
+
+  beforeEach(async () => {
+    // Deploying from a simple account, as proxy won't call on it in this test.
+    stakerProxy = await StakerProxy.new(
+      owner,
+      { from: composer },
+    );
+
+    spyGateway = await SpyGateway.new();
+  });
+
+  it('should approve and stake on the gateway', async () => {
+    const spyValueToken = await SpyToken.at(await spyGateway.valueToken.call());
+    const spyBaseToken = await SpyToken.at(await spyGateway.baseToken.call());
+
+    await stakerProxy.stake(
+      amount,
+      beneficiary,
+      gasPrice,
+      gasLimit,
+      nonce,
+      hashLock,
+      spyGateway.address,
+      { from: composer },
+    );
+
+    assert.strictEqual(
+      (await spyGateway.amount.call()).toString(10),
+      amount,
+      'The spy did not record the correct amount staked.',
+    );
+
+    assert.strictEqual(
+      (await spyGateway.beneficiary.call()),
+      beneficiary,
+      'The spy did not record the correct beneficiary.',
+    );
+
+    assert.strictEqual(
+      (await spyGateway.gasPrice.call()).toString(10),
+      gasPrice,
+      'The spy did not record the correct gas price staked',
```

Comment:
Full stop can be added in the end of error messages 👁 . Similarly for others messages .